### ISSUE TITLE
[0.4.4] restrict-plus-operands misses any callback indices after chained map or slice calls

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1271]
## Summary

Follow-up to #1260: its direct `items.map(...)` reproduction is fixed in 0.4.4. However, when `items: any` first goes through `.slice(0)` or `.map(...)`, a subsequent `.map((value, index) => index + 1)` still fails to report the any operand with explicit `allowAny: false`.

Since the receiver is any, these calls and callback parameters remain untyped. In contrast, the same chain on a number[] receiver correctly has a numeric index and should not report.

Verified with @utoo/lint 0.4.4, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, Node.js 22.23.1 on macOS arm64. Only this rule is enabled; there are no parsing failures or suppression comments. Explicit allowAny:false avoids differences in defaults between plugin versions.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.4 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0 typescript@5.3.3
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

Run each snippet separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## chain-direct-any

```tsx
export function example(items:any){return items.map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## chain-any-map-map

```tsx
export function example(items:any){return items.map(value=>value).map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## chain-any-slice-map

```tsx
export function example(items:any){return items.slice(0).map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## chain-any-map-slice-map

```tsx
export function example(items:any){return items.map(value=>value).slice(0).map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## chain-array-control

```tsx
export function example(items:number[]){return items.map(value=>value).slice(0).map((value,index)=>index+1);}
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

Expected: report the three any-receiver chains just like the already-fixed direct-call control. Preserve the clean number[] control.
